oak-run/src/main/java/org/apache/jackrabbit/oak/index/merge/IndexDefMergerUtils.java

```diff
@@ -148,6 +148,8 @@ private static String mergeProperty(String path, String property, JsonObject anc
             return cp;
         } else if (Objects.equals(ap, cp)) {
             return pp;
+        } else if (ap == null) {
+            return cp;
```

Comment:
It would be good to add a comment, e.g.

    // if it doesn't exist in the ancestor, prefer customization over the product value

However, I'm not quite sure why this behavior is needed... shouldn't we prefer pp (product)? Or maybe cp == pp ?

Reply:
My thought is that if the custom index defined the _same_ property name with a different value as is now in the product index, but that property was not present before, that we should preserve that custom value as the developer of the custom index would expect their customizations to be preserved not undone.